includes/class-wcpdf-main.php

```diff
@@ -444,8 +444,15 @@ public function generate_document_ajax() {
 				}
 				break;
 			case 'full':
-				// no order
-				if ( ! $order ) {
+				// check the user privileges
+				$full_permission = WPO_WCPDF()->admin->user_can_manage_document( $document_type );
```

Comment:
I'm not sure if this is correct, because the `user_can_manage_document` checks for this:

`current_user_can( 'manage_woocommerce_orders' ) || current_user_can( 'edit_shop_orders' )`

which means that `guest` access will not work here.

The full intent is to allow anyone with the link to have access, I think this way is more restrictive.

I think the issue is related to the Bulk Action not passing the access key in the URL.

Reply:
> which means that guest access will not work here.

May I ask why it won't work? I'm a little confused.
Since the condition I added will break the `case` if there is full permission. If the access type is "guest" and the user is not logged in, the `$full_permission` is also `false`, so it won't break the `case` and should work as before.

> I think the issue is related to the Bulk Action not passing the access key in the URL.

Yes, the access key doesn't exist in the URL, but `_wpnonce` exists, and this block of code fills the `access_key`:
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/21e59a7a35c23d700870da9bd17c7820370e12e9/includes/class-wcpdf-main.php#L330

After that, it gets to this block of code. It won't execute for bulk action, and the `$order` will always be `false`:
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/21e59a7a35c23d700870da9bd17c7820370e12e9/includes/class-wcpdf-main.php#L383

So I think the issue was the condition in the full case that checked if the `$order` is not true (`if ( ! $order )`).
Am I right?

Reply:
I think you're right, I haven't check the `access_key` part in the code, didn't recall that existed, and ignored all the rest because without the access key the below code doesn't execute, sorry.